src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -1186,4 +1182,94 @@ private void sendBulkUploadNotification(String orgId, String orgName, String fil
 				serverConfig.getSbUrl() + serverConfig.getSbSendNotificationEmailPath(), request,
 				ProjectUtil.getDefaultHeaders());
 	}
+
+	@Override
+	public SBApiResponse getNotificationPreferencesById(String userId) {
+		SBApiResponse response = ProjectUtil.createDefaultResponse(Constants.API_READ_NOTIFICATION_PREFERENCE);
+		String errMsg = null;
+		if (StringUtils.isEmpty(userId)) {
+			response.getParams().setErrmsg("Please Provide a user Id");
+			response.setResponseCode(HttpStatus.BAD_REQUEST);
+			response.getParams().setStatus(Constants.FAILED);
+			return response;
+		}
+		Map<String, Object> request = new HashMap<>();
+		request.put(Constants.USER_ID, userId);
+		try {
+			List<Map<String, Object>> notificationPreferences = cassandraOperation.getRecordsByProperties(
+					Constants.KEYSPACE_SUNBIRD, Constants.TABLE_USER_NOTIFICATION_PREFERENCE, request,
+					Collections.singletonList(Constants.NOTIFICATION_PREFERENCE));
+			GsonBuilder builder = new GsonBuilder();
+			Gson gson = builder.create();
+			List<Map<String, Object>> mapping = gson.fromJson(notificationPreferences.toString(), List.class);
+			for (Map<String, Object> responseMap : mapping) {
+				response.putAll(responseMap);
+			}
+			response.setResponseCode(HttpStatus.OK);
+			response.getParams().setStatus(Constants.SUCCESSFUL);
+		} catch (Exception e) {
+			errMsg = "Failed to process message. Exception: " + e.getMessage();
+		}
+		if (StringUtils.isNotBlank(errMsg)) {
+			response.getParams().setStatus(Constants.FAILED);
+			response.getParams().setErrmsg(errMsg);
+			response.setResponseCode(HttpStatus.BAD_REQUEST);
+		}
+		return response;
+	}
+
+	@Override
+	public SBApiResponse updateNotificationPreference(String userId, Map<String, Object> request) {
+		SBApiResponse response = ProjectUtil.createDefaultResponse(Constants.API_UPDATE_NOTIFICATION_PREFERENCE);
+		String errMsg = null;
+		if (StringUtils.isEmpty(userId)) {
+			response.getParams().setErrmsg("Please Provide user Id");
+			response.setResponseCode(HttpStatus.BAD_REQUEST);
+			response.getParams().setStatus(Constants.FAILED);
+			return response;
+		}
+		try {
+			Map<String,Object> requestBody= (Map<String, Object>) request.get(Constants.REQUEST);
+			Map<String, Object> updateRequest = new HashMap<>();
+			if(MapUtils.isNotEmpty(requestBody)){
+				ObjectMapper objectMapper = new ObjectMapper();
+				String json = objectMapper.writeValueAsString(request);
+				updateRequest.put(Constants.NOTIFICATION_PREFERENCE, json);
+			}
+			Map<String, Object> Id= new HashMap<String, Object>() {
```

Comment:
Variable name should not start with capital letter... name this variable as "key"